hw3_2/hw3-2/Program.cs

```diff
@@ -0,0 +1,452 @@
+                    if (i == textForCompressionLenght - 1) //if last suffix is new 
+                    {
+                        codesOfCompressedText.Add(BitConverter.GetBytes(Convert.ToUInt32(currentCode)));
+                    }
+                }
+            }
+
+            return codesOfCompressedText;
+        }
+
+        public string LzwDecompression(List<Byte[]> compressedMessage, out bool isEmptyText, out bool isCorrectText)
+        {
+            isCorrectText = true;
+            string decompressedText = "";
+            int amountOfSuffixesInCompressedText = compressedMessage.Count;
+            if (amountOfSuffixesInCompressedText == 0)
+            {
+                isEmptyText = true;
+                return "";
+            }
+
+            isEmptyText = false;
+            List<string> suffixes = new();
+            string lastSuffix = ""; //unnecessary initialization
+            AddAllSymbolsToList(suffixes);
+            for (int i = 0; i < amountOfSuffixesInCompressedText; i++)
+            {
+                UInt32 uIntCode = BitConverter.ToUInt32(compressedMessage[i]);
+                int intCode = Convert.ToInt32(uIntCode);
+                if (intCode < suffixes.Count)
+                {
+                    decompressedText += suffixes[intCode];
+                    if (i > 0)
+                    {
+                        suffixes.Add(lastSuffix +
+                                     suffixes[intCode][0]); //there is warning here if no initialization
+                    }
+
+                    lastSuffix = suffixes[intCode];
+                }
+                else if (intCode == suffixes.Count)
+                {
+                    lastSuffix = lastSuffix + lastSuffix[0];
+                    suffixes.Add(lastSuffix);
+                    decompressedText += lastSuffix;
+                }
+                else
+                {
+                    isCorrectText = false;
+                    return "";
+                }
+            }
+
+            return decompressedText;
+        }
+    }
+
+    class BorStringStorage
+    {
+        class Vertex
+        {
+            public int NumberOfSuffix;
+            public int HowManyDescendants;
+            public bool IsTerminal;
+            public Vertex[] ArrayOfRelatedVertexes;
+
+            public Vertex(bool isTerminal, Vertex[] arrayOfRelatedVertexes)
+            {
+                NumberOfSuffix = -1;
+                HowManyDescendants = 0;
+                this.IsTerminal = isTerminal;
+                this.ArrayOfRelatedVertexes = arrayOfRelatedVertexes;
+            }
+        }
+
+        private int _size = 0;
+
+        public BorStringStorage()
+        {
+            Vertex[] currArrayOfVertexes = new Vertex[256];
+            _root = new Vertex(false, currArrayOfVertexes);
+        }
+
+        public int Size {
+            get => _size;
+            private set { _size = value; }
+        }

```

Comment:
```suggestion
        public int Size {
            get;
            private set;
        }
```
Вы написали абсолютно стандартную реализацию для `get` и `set`. Для такого случая есть более краткая форма.
Такая форма также неявно создаст приватное поле, где будет храниться значение.